gammapy/background/models.py

```diff
@@ -71,3 +77,491 @@ def evaluate(self, x, y):
+                ii = det_bin[0]
+            print("ii", ii)
+            for jj in range(n_det_bins_y):
+                if do_only_1_plot:
+                    jj = det_bin[1]
+                print(" jj", jj)
+                data = self.background[:, ii, jj]
+                detx_bin_center = det_bin_centers[0, ii]
+                dety_bin_center = det_bin_centers[0, jj]
+                print ( "  image({}) canvas({}) pad({})".format(count_images, count_canvases, count_pads))
+
+                if do_only_1_plot:
+                    fig.set_size_inches(8., 8., forward=True)
+                    ax = axes
+                else:
+                    ax = axes.flat[count_pads - 1]
+                image = ax.plot(energy_points.to('TeV'), data, drawstyle='default') # connect points with lines
+                ax.loglog() # double log scale # slow!
+                if do_only_1_plot:
+                    ss_detx_bin_edges = "[{0:.1f}, {1:.1f}) {2}".format(det_bin_edges[0].value, det_bin_edges[1].value, det_bin_edges.unit)
+                    ss_dety_bin_edges = "[{0:.1f}, {1:.1f}) {2}".format(det_bin_edges[2].value, det_bin_edges[3].value, det_bin_edges.unit)
+
+                    ax.set_title('Det = {0} {1}'.format(ss_detx_bin_edges, ss_dety_bin_edges))
+                else:
+                    ss_det_bin_center = "({0:.1f}, {1:.1f})".format(detx_bin_center, dety_bin_center)
+                    ax.set_title('Det = {}'.format(ss_det_bin_center))
+                ax.set_xlabel('E / {}'.format(energy_points.unit))
+                ax.set_ylabel('Bg rate / {}'.format(data.unit))
+                count_pads += 1 # increase
+
+                if count_pads > npads_per_canvas or count_images == nimages:
+                    print("Canvas full, saving and creating a new canvas")
+                    if do_only_1_plot:
+                        filename = "cube_background_model_spectrum{0}{2}{1}{2}.png".format(det.value[0], det.value[1], det.unit)
+                    else:
+                        filename = "cube_background_model_spectra{}.png".format(count_canvases)
+                    print('Writing {}'.format(filename))
+                    fig.savefig(filename)
+                    if not do_only_1_plot:
+                        plt.close('all') # close all open figures
+                        fig, axes = plt.subplots(nrows=nrows, ncols=ncols)
+                        fig.set_size_inches(25., 25., forward=True)
+                    count_canvases += 1 # increase
+                    count_pads = 1 # reset
+
+                count_images += 1 # increase
+
+        if do_only_1_plot:
+            return fig, axes, image
+
+
+    def write(self, filename):
+        """Write cube background model to fits file.
+
+        Parameters
+        ----------
+        filename : `~string`
+        	name of file for the bg cube
+        """
+        hdu = fits.ImageHDU(data=self.background.value)
+        # TODO: save binning also!!!!
```

Comment:
It would be nice if there was a separate `write` method that stores the background cube as a 3D FITS IMAGE extension. And if you manage to create a WCS object representing the DETX / DETY binning, ds9 could show it with correct values of DETX and DETY as world coordinates.

I'm pretty sure this is possible, and it would be nice to have, but as I said, I would have to research / play around with this myself, so if you don't know where to start or find something within ~ 1 hour, leave this part to me.